packages/flutter/lib/src/rendering/sliver_persistent_header.dart

```diff
@@ -595,12 +601,18 @@ abstract class RenderSliverFloatingPersistentHeader extends RenderSliverPersiste
     final double maxExtent = this.maxExtent;
     final double paintExtent = maxExtent - _effectiveScrollOffset!;
```

Comment:
I'm currently investigating the cause of the demo's bar shaking.
1，Here, we calculate the insets depended on the scroll offset, but `updateGeometry` only be called when layout occurs.
     If the user scrolling without perform layout(you know this always happens), the content inset will not updated. Once the layout occurs after more scrolling, the content insets will updated and cause the bar shaking.

It seems that we also should update the geometry when scroll position changed.

Reply:
![image](https://user-images.githubusercontent.com/61075224/186812595-5c497ce6-ca2b-4c5d-916f-d4434b5e9a95.png)

Reply:
### Demo (Tested on desktop to avoid status bar issues.)
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
          body: RawScrollbar(
            thumbVisibility: true,
            thickness: 20,
            thumbColor: Colors.red,
            child: ScrollConfiguration(
              behavior: ScrollConfiguration.of(context).copyWith(scrollbars: false),
              child: CustomScrollView(
                primary: true,
                slivers: [
                  const SliverAppBar(
                    floating: true,
                    toolbarHeight: 100,
                    // pinned: true,
                    // snap: true,
                    // stretch: true,
                  ),
                  SliverList(
                      delegate: SliverChildBuilderDelegate(
                            (context, index) => Text('Item $index'),
                        childCount: 200,
                      )
                  ),
                ],
              ),
            ),
          )
      ),
    );
  }
}
```